src/runtime-rs/crates/shimv2/src/shim_start.rs

```diff
@@ -0,0 +1,218 @@
+        Ok(())
+    }
+
+    fn do_start(&mut self) -> Result<PathBuf> {
+        let spec = self.load_oci_spec()?;
+        let info = get_shim_id_info(&spec).context("container sandbox info")?;
+
+        match info {
+            ShimIdInfo::Sandbox => {
+                let address = self.socket_address(&self.args.id)?;
+                let socket = new_listener(&address)?;
+                let child_pid = self.create_shim_process(socket)?;
+                self.write_pid_file(child_pid)?;
+                self.write_address(&address)?;
+                Ok(address)
+            }
+            ShimIdInfo::Container(sandbox_id) => {
+                let (address, pid) = self.get_shim_info_from_sandbox(&sandbox_id)?;
+                self.write_pid_file(pid)?;
+                self.write_address(&address)?;
+                Ok(address)
+            }
+        }
+    }
+
+    fn new_command(&self) -> Result<std::process::Command> {
+        if self.args.id.is_empty()
+            || self.args.namespace.is_empty()
+            || self.args.address.is_empty()
+            || self.args.publish_binary.is_empty()
+        {
+            return Err(anyhow!("invalid param"));
+        }
+
+        let bundle_path = self.get_bundle_path()?;
+        let self_exec = std::env::current_exe().map_err(Error::SelfExec)?;
+        let mut command = std::process::Command::new(self_exec);
+
+        command
+            .current_dir(bundle_path)
+            .stdin(std::process::Stdio::null())
+            .stdout(std::process::Stdio::null())
+            .stderr(std::process::Stdio::null())
+            .arg("-id")
+            .arg(&self.args.id)
+            .arg("-namespace")
+            .arg(&self.args.namespace)
+            .arg("-address")
+            .arg(&self.args.address)
+            .arg("-publish-binary")
+            .arg(&self.args.publish_binary)
+            .env("RUST_BACKTRACE", "1");
+
+        if self.args.debug {
+            command.arg("-debug");
+        }
+
+        Ok(command)
+    }
+
+    fn create_shim_process<T: IntoRawFd>(&self, socket: T) -> Result<u32> {
+        let mut cmd = self.new_command()?;
+        cmd.env(ENV_KATA_BIND_FD, format!("{}", socket.into_raw_fd()));
+        let child = cmd.spawn().map_err(Error::SpawnChild)?;
+
+        Ok(child.id())
+    }
+
+    fn get_shim_info_from_sandbox(&self, sandbox_id: &str) -> Result<(PathBuf, u32)> {
+        // All containers of a pod share the same pod socket address.
+        let address = self.socket_address(sandbox_id)?;
+        let bundle_path = self.get_bundle_path()?;
+        let sandbox_bundle_path = bundle_path
+            .join("..")
+            .join(sandbox_id)
+            .canonicalize()
+            .context(Error::GetBundlePath)?;
+        let pid = self.read_pid_file(&sandbox_bundle_path)?;
+
+        Ok((address, pid))
+    }
+}
+
+fn new_listener<T: AsRef<OsStr>>(address: T) -> Result<UnixListener> {
+    // Listen on an abstract Unix Domain Socket.
```

Comment:
Does still need to use an abstract Unix domain socket? @lifupan

Reply:
Thanks for your review, has changed to the normal Unix domain socket(same as the golang version)